spec/graphql/static_validation/rules/unique_directives_per_location_spec.rb

```diff
@@ -0,0 +1,180 @@
+require "spec_helper"
+
+describe GraphQL::StaticValidation::UniqueDirectivesPerLocation do
+  include StaticValidationHelpers
+
+  let(:schema) { GraphQL::Schema.from_definition("
+    type Query {
+      type: Type
+    }
+
+    type Type {
+      field: String
+    }
+
+    directive @A on FIELD
+    directive @B on FIELD
+  ") }
+
+  describe "query with no directives" do
+    let(:query_string) {"
+      {
+        type {
+          field
+        }
+      }
+    "}
+
+    it "passes rule" do
+      assert_equal [], errors
+    end
+  end
+
+  describe "query with unique directives in different locations" do
+    let(:query_string) {"
+      {
+        type @A {
+          field @B
+        }
+      }
+    "}
+
+    it "passes rule" do
+      assert_equal [], errors
+    end
+  end
+
+  describe "query with unique directives in same locations" do
+    let(:query_string) {"
+      {
+        type @A @B {
+          field @A @B
+        }
+      }
+    "}
+
+    it "passes rule" do
+      assert_equal [], errors
+    end
+  end
+
+  describe "query with same directives in different locations" do
+    let(:query_string) {"
+      {
+        type @A {
+          field @A
+        }
+      }
+    "}
+
+    it "passes rule" do
+      assert_equal [], errors
+    end
+  end
+
+  describe "query with same directives in similar locations" do
+    let(:query_string) {"
+      {
+        type {
+          field @A
+          field @A
+        }
+      }
+    "}
+
+    it "passes rule" do
+      assert_equal [], errors
+    end
+  end
+
+  describe "query with duplicate directives in one location" do
+    let(:query_string) {"
+      {
+        type {
+          field @A @A
+        }
+      }
+    "}
+
+    it "fails rule" do
+      assert_includes errors, {
+        "message" => 'The directive "A" can only be used once at this location.',
+        "locations" => [{ "line" => 4, "column" => 17 }, { "line" => 4, "column" => 20 }],
+        "fields" => ["query", "type", "field"],
```

Comment:
It was [renamed to disambiguate from runtime errors](https://github.com/rmosolgo/graphql-ruby/pull/264)